app/questionnaire/navigator.py

```diff
@@ -33,14 +33,45 @@ def evaluate_goto(goto_rule, answers, group_instance):
     :return:
     """
     if 'when' in goto_rule.keys():
-        answer_index = goto_rule['when']['id']
-        filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
-        if len(filtered) == 1:
-            return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        if 'id' in goto_rule['when']:
+            answer_index = goto_rule['when']['id']
+            filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
+            if len(filtered) == 1:
+                return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        if 'meta' in goto_rule['when']:
+            return evaluate_metadata_rule(goto_rule, metadata)
+
         return False
     return True
 
 
+def evaluate_metadata_rule(goto_rule, metadata):
+    meta_index = goto_rule['when']['meta']
+    if _contains_in_dict(metadata, meta_index):
+        metadata_value = _get_from_dict(metadata, meta_index)
+        return evaluate_rule(goto_rule, metadata_value)
+
+
+def _get_from_dict(d, keys):
+    if "." in keys:
```

Comment:
for the region it is defined at the root of the json but the variant_flags are nested so we need both scenarios